### PR TITLE
Publish harvester-cluster-repo image

### PR DESCRIPTION
**Pull request by bk201**
Related issue: https://github.com/harvester/harvester/issues/2670

The image is published when:
- merging codes to master and stable branches in the harvester-installer repo (this PR).
- cron CI job runs in the harvester-installer repo (this PR).
- merging codes to master and stable branches in the harvester/harvester repo (https://github.com/harvester/harvester/pull/4223).

### CODE REVIEW

**Review by guangbochen — COMMENTED** [review]
This is not a big deal but in my build there is a chart repo in the `dist/harvester-cluster-repo` which is not expected.
```
hp-163-seeder:/home/rancher/src/harvester-installer/dist/harvester-cluster-repo # ls -la
total 4
drwxr-xr-x 1 root root  32 Jul 13 19:05 .
drwxr-xr-x 1 root root  62 Jul 13 19:05 ..
-rw-r--r-- 1 root root 401 Jul 13 19:05 Dockerfile
drwxr-xr-x 1 root root 718 Jul 13 19:05 charts
hp-163-seeder:/home/rancher/src/harvester-installer/dist/harvester-cluster-repo # ls -la charts/
total 1124
drwxr-xr-x 1 root root    718 Jul 13 19:05 .
drwxr-xr-x 1 root root     32 Jul 13 19:05 ..
-rw-r--r-- 1 root root 170784 Jul 13 19:05 harvester-0.0.0-master-b0530a4d.tgz
-rw-r--r-- 1 root root  56005 Jul 13 19:05 harvester-crd-0.0.0-master-b0530a4d.tgz
-rw-r--r-- 1 root root   2683 Jul 13 19:05 harvester-pcidevices-controller-0.2.5-rc2.tgz
-rw-r--r-- 1 root root   2914 Jul 13 19:05 harvester-seeder-0.1.0-rc2.tgz
-rw-r--r-- 1 root root   2635 Jul 13 19:05 harvester-vm-import-controller-0.1.4.tgz
-rw-r--r-- 1 root root  11870 Jul 13 19:05 index.yaml
-rw-r--r-- 1 root root  14652 Jul 13 19:05 rancher-logging-102.0.0+up3.17.10.tgz
-rw-r--r-- 1 root root  83284 Jul 13 19:05 rancher-logging-crd-102.0.0+up3.17.10.tgz
-rw-r--r-- 1 root root 402958 Jul 13 19:05 rancher-monitoring-102.0.0+up40.1.2.tgz
-rw-r--r-- 1 root root 387142 Jul 13 19:05 rancher-monitoring-crd-102.0.0+up40.1.2.tgz
```

File: .drone.yml [hunk 1]
```diff
@@ -39,6 +39,49 @@ steps:
...
+    username:
+      from_secret: docker_username
+    password:
+      from_secret: docker_password
+  when:
+    instance:
+    - drone-publish.rancher.io
+    branch:
+    - master
+    - v*
```
[the remark this answers is not in the document]
Comment (bk201) [reply .drone.yml]: Fixed.

**Review by futuretea — APPROVED** [review]
LGTM, thanks.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Comment by guangbochen** [comment]
@mergify backport v1.2

**Comment by bk201** [comment]
@mergifyio backport v1.1

**Pull request merged**